pySDC/projects/PinTSimE/discontinuous_test_ODE.py

```diff
@@ -11,6 +11,8 @@
 from pySDC.projects.PinTSimE.battery_model import get_recomputed, generate_description
 import pySDC.helpers.plot_helper as plt_helper
 from pySDC.core.Hooks import hooks
+from pySDC.implementations.hooks.log_errors import LogLocalErrorPostStep
```

Comment:
Ok, I see that this is poorly implemented on my part. The local error requires an exact solution starting from the beginning of the step, not from t=0. That's why some of the `u_exact` functions have the `u_init` and `t_init` variables. However, if they are not used, the `LogLocalErrorPostStep` function logs the global error without telling you. Maybe we should put in warnings whenever this happens...

Reply:
Hm, I would add a warning now in the discontinuous test problem class, but in this case a warning does make sense for all. How would you add the warning?

Reply:
I can also put a warning this in the other problem classes where t_init and u_init is set the parameters.

Reply:
I added it to the advection, heat and nonlinear Schrödinger equations, but I am waiting for the tests to pass before I start the PR.
[here](https://github.com/brownbaerchen/pySDC/blob/fa2fa12962989a6b4a1a4c5857e2cd0888b861e6/pySDC/implementations/problem_classes/AdvectionEquation_ND_FD.py#L100) is the warning I added. It doesn't need to be too fancy, I guess. Note that in the function definition, are only the variables that are used in the function and then `**kwargs` to allow for additional keyword arguments. I don't know if this is the pythonic way, but I think this is nice because you can more easily tell by the function definition what arguments are allowed.
If you pass on the keyword arguments to another function, a good editor will show the allowed arguments for the both functions when you look up the documentation for the "outer" function. Although I have to admit my editor struggles with this sometimes. ;)

Reply:
Of course the `kwargs**` argument is very useful! I would add a warning for `AllenCahn_2D_FFT`, `DiscontinuousTestODE`, `Lorenz`, `Piline`, `Quench`, `TestEquation_0D` and `Van_der_Pol_implicit`, which I already did but will update the warning.

Reply:
Are you sure non of them can compute the local error? I looked up the test equation and even though it's analytic, it can compute the local error. All other problems you mentioned rely on numerical exact solutions, I believe, which should support the local error.